lms/resources/lti_launch.py

```diff
@@ -82,6 +82,19 @@ def h_group(self):
     def _is_speed_grader(self):
         return bool(self._request.GET.get("learner_canvas_user_id"))
 
+    @property
+    def is_legacy_speed_grader(self):
+        # Detect speed grader launches without the fix for
+        #
+        # * https://github.com/instructure/canvas-lms/issues/1952
+        # * https://github.com/hypothesis/lms/issues/3228
+        #
+        # applied which includes the correct value for `resource_link_id` in the query params.
```

Comment:
I think this is a docstring rather than a comment. Maybe something like:

```python
        """
        Return True if the current request is a legacy SpeedGrader launch.

        To work around a Canvas bug we add the assignment's resource_link_id as
        a query param on the LTI launch URLs that we submit to SpeedGrader (see
        https://github.com/instructure/canvas-lms/issues/1952 and
        https://github.com/hypothesis/lms/issues/3228).

        "Legacy" SpeedGrader submissions are ones from before we implemented
        this work-around, so they don't have the resource_link_id query param.
        """
```

Always good to be explicit in the first line of a docstring ("Return True if..." rather then "Detect...").